src/nurse_ussd.js

```diff
@@ -175,558 +152,359 @@ go.app = function() {
+                            .opt_in(self.im, self.contact)
+                            .then(function() {
+                                return 'st_faccode';
+                            });
+                    } else {
+                        return 'st_stay_out';
+                    }
+                }
             });
         });
 
-        self.add('states_id_type', function(name) {
+        self.add('st_stay_out', function(name) {
             return new ChoiceState(name, {
-                question: $('What kind of identification does the pregnant ' +
-                            'mother have?'),
-
+                question: $("You have chosen not to receive NurseConnect SMSs on this number and so cannot complete registration."),
                 choices: [
-                    new Choice('sa_id', $('SA ID')),
-                    new Choice('passport', $('Passport')),
-                    new Choice('none', $('None'))
+                    new Choice('isl_route', $('Main Menu'))
                 ],
-
                 next: function(choice) {
-                    self.contact.extra.id_type = choice.value;
-
-                    return self.im.contacts
-                        .save(self.contact)
-                        .then(function() {
-                            return {
-                                sa_id: 'states_sa_id',
-                                passport: 'states_passport_origin',
-                                none: 'states_birth_year'
-                            } [choice.value];
-                        });
+                    return choice.value;
                 }
             });
         });
 
-        self.add('states_sa_id', function(name, opts) {
-            var error = $('Sorry, the mother\'s ID number did not validate. ' +
-                          'Please reenter the SA ID number:');
-
-            var question = $('Please enter the pregnant mother\'s SA ID ' +
-                            'number:');
-
+        self.add('st_faccode', function(name) {
+            var owner = self.user.extra.working_on === "" ? 'your' : 'their';
+            var error = $("Sorry, that code is not recognized. Please enter the 6-digit facility code again, e. 535970:");
+            var question = $("Please enter {{owner}} 6-digit facility code:")
+                .context({owner: owner});
             return new FreeText(name, {
                 question: question,
-
                 check: function(content) {
-                    if (!go.utils.validate_id_sa(content)) {
-                        return error;
-                    }
+                    return go.utils
+                        .validate_clinic_code(self.im, content.trim())
+                        .then(function(facname) {
+                            if (!facname) {
+                                return error;
+                            } else {
+                                self.contact.extra.facname = facname;
```

Comment:
as a general, all app thing, it might be better to namespace all the nurseconnect extras as `nc_` for protection and easier debug